addons/mail/controllers/common.py

```diff
@@ -0,0 +1,51 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import http
+from odoo.http import request
+
+
+class MailControllerCommon(http.Controller):
```

Comment:
Name file according to content, like ``mail_controller.py`` (common is more something used in tests, but don't think we have to add "common" to all inherited things).

Reply:
We already have `MailController` and `mail.py` in the portal controllers. Isn't it confusing to change these names to not include `common`?

Reply:
``DiscussController`` ? Indeed ``MailController`` holds the "mail" part with record redirection and stuff.

Reply:
Or even keep ``ThreadController`` which was fine.

Reply:
We don't need to use "common" each time we use standard class inheritance :p

Reply:
And you don't have to move the code, keep it in thread.py. You can have two classes in the same file, less diff, less annoying blame when searching in files, ...

(sorry for spam)

Reply:
So I guess I keep the `ThreadController` as it was before and inherit from it when it's needed (In AttachmentController, MessageReactionController, WebclientController).

Reply:
Less diff, happy blame !